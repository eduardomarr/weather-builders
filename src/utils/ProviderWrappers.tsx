import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { WeatherContext } from '../context/weather';
import theme from '../styles/theme';

type WrapperProps = {
  children: React.ReactNode;
};

const mockedContextValues = {
  weather: {
    description: 'description',
    temp: 10,
    feelsLike: 16,
    tempMin: 10,
    tempMax: 16,
    pressure: 10,
    humidity: 50,
    windSpeed: 50,
    district: 'SÃ£o Paulo',
    visibility: 'visibility',
    sunset: 'sunset',
    sunrise: 'sunrise',
    icon: 'icon',
  },

  getCurrentWeather: jest.fn(),
  loading: false,
  setLoading: jest.fn(),
};

export const ThemeWrapper: React.FC<WrapperProps> = ({ children }) => {
  return (
      <ThemeProvider theme={theme}>
        {children}
      </ThemeProvider>
  );
};

export const ContextWrapper: React.FC<WrapperProps> = ({ children }) => (
  <WeatherContext.Provider value={mockedContextValues}>
    {children}
  </WeatherContext.Provider>
);
