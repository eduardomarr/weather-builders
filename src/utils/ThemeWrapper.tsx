import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import theme from '../styles/theme';

type WrapperProps = {
  children: React.ReactNode;
};

export const ThemeWrapper: React.FC<WrapperProps> = ({ children }) => {
  return (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
};
