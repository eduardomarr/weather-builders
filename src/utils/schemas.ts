import * as Yup from 'yup';

export const signInSchema = Yup.object().shape({
  email: Yup.string().required('E-mail é obrigatório').email('E-mail Inválido!'),
  password: Yup.string().required('Senha é obrigatório'),
});
