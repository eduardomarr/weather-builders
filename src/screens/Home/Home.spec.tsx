import React from 'react';

import { Home } from '.';
import { render } from '@testing-library/react-native';
import { ContextWrapper, ThemeWrapper } from '../../utils/ProviderWrappers';
import renderer from 'react-test-renderer';


jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => {
      return {
        navigate: jest.fn(),
      };
    },
  };
});

const HomeScreenWrapped = () => {
  return (
    <ContextWrapper>
      <ThemeWrapper>
        <Home />
      </ThemeWrapper>
    </ContextWrapper>
  );
};

describe('Home Screen', () => {

  it('should be render correctly', () => {

    const {getByTestId} = render(
      <HomeScreenWrapped />
    );

    expect(getByTestId('container')).toBeOnTheScreen();
  });

  it('should be render main card Current Weather', () => {

    const {getByTestId} = render(
      <HomeScreenWrapped />
    );

    expect(getByTestId('container')).toBeOnTheScreen();
    expect(getByTestId('district')).toBeOnTheScreen();
    expect(getByTestId('current-temp')).toBeOnTheScreen();
    expect(getByTestId('icon')).toBeOnTheScreen();
    expect(getByTestId('description')).toBeOnTheScreen();
    expect(getByTestId('min-temp')).toBeOnTheScreen();
    expect(getByTestId('max-temp')).toBeOnTheScreen();

  });

  it('should be render Cards Grid Weather', () => {

    const {getByTestId} = render(
      <HomeScreenWrapped />
    );

    expect(getByTestId('card-grid')).toBeOnTheScreen();

  });

  it('should be render today value', () => {

    const {getByTestId} = render(
      <HomeScreenWrapped />
    );

    expect(getByTestId('today')).toBeOnTheScreen();
  });
});

