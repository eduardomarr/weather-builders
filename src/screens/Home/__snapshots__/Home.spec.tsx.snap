// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen should be styled correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4278782258,
      4279994086,
    ]
  }
  endPoint={
    Object {
      "x": 1.5,
      "y": 1.8,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.1,
      "y": 0.9,
    }
  }
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        colors={
          Array [
            "#FFFFFF",
          ]
        }
        onRefresh={[Function]}
        progressBackgroundColor="#1B86E6"
        refreshing={false}
        tintColor="#FFFFFF"
      />
    }
    showsVerticalScrollIndicator={false}
    style={
      Array [
        Object {},
      ]
    }
    testID="container"
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingBottom": 56,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 56,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "marginBottom": 24,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={55}
              bbWidth={55}
              fill="#FFFFFF"
              focusable={false}
              height={55}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Array [
                    Object {},
                  ],
                  Object {
                    "flex": 0,
                    "height": 55,
                    "width": 55,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
              width={55}
              xmlns="http://www.w3.org/2000/svg"
            >
              <RNSVGGroup
                fill={4294967295}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M0 0h24v24H0z"
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M5.334 4.545a9.99 9.99 0 013.542-2.048A3.993 3.993 0 0012 3.999a3.993 3.993 0 003.124-1.502 9.99 9.99 0 013.542 2.048 3.993 3.993 0 00.262 3.454 3.993 3.993 0 002.863 1.955 10.043 10.043 0 010 4.09c-1.16.178-2.23.86-2.863 1.955a3.993 3.993 0 00-.262 3.455 9.99 9.99 0 01-3.542 2.047A3.993 3.993 0 0012 20a3.993 3.993 0 00-3.124 1.502 9.99 9.99 0 01-3.542-2.047A3.993 3.993 0 005.072 16a3.993 3.993 0 00-2.863-1.954 10.043 10.043 0 010-4.091A3.993 3.993 0 005.072 8a3.993 3.993 0 00.262-3.454zM13.5 14.597a3 3 0 10-3-5.196 3 3 0 003 5.196z"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#222149",
                "borderRadius": 16,
                "justifyContent": "flex-start",
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Manrope-Medium",
                  "fontSize": 39,
                  "textAlign": "center",
                },
              ]
            }
            testID="district"
          >
            São Paulo
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Manrope-Bold",
                  "fontSize": 63,
                  "marginBottom": 16,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 16,
                  "textAlign": "center",
                },
              ]
            }
            testID="current-temp"
          >
            10
            °
          </Text>
          <Image
            source={
              Object {
                "uri": "icon",
              }
            }
            style={
              Array [
                Object {
                  "height": 255,
                  "width": 255,
                },
              ]
            }
            testID="icon"
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Manrope-Regular",
                  "fontSize": 31,
                  "marginBottom": 16,
                  "textAlign": "center",
                },
              ]
            }
            testID="description"
          >
            description
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Manrope-Regular",
                    "fontSize": 24,
                    "marginBottom": 4,
                    "marginLeft": 8,
                    "marginRight": 8,
                    "marginTop": 4,
                    "textAlign": "center",
                  },
                ]
              }
              testID="min-temp"
            >
              min: 
              10
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "height": "80%",
                    "width": 1,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Manrope-Regular",
                    "fontSize": 24,
                    "marginBottom": 4,
                    "marginLeft": 8,
                    "marginRight": 8,
                    "marginTop": 4,
                    "textAlign": "center",
                  },
                ]
              }
              testID="max-temp"
            >
              max: 
              16
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Manrope-Regular",
                "fontSize": 31,
                "marginBottom": 32,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 32,
                "textAlign": "center",
              },
            ]
          }
          testID="today"
        >
          sexta-feira, 6 de janeiro de 2023
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "height": 883,
                "paddingBottom": 80,
              },
            ]
          }
          testID="card-grid"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#222149",
                  "borderRadius": 16,
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "maxHeight": 150,
                  "minWidth": 150,
                  "paddingBottom": 4,
                  "paddingLeft": 4,
                  "paddingRight": 4,
                  "paddingTop": 4,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Manrope-SemiBold",
                    "fontSize": 35,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Pressão
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": "100%",
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={47}
                bbWidth={47}
                fill="#FFFFFF"
                focusable={false}
                height={47}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {},
                    ],
                    Object {
                      "flex": 0,
                      "height": 47,
                      "width": 47,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={47}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M0 0h24v24H0z"
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm1-10V7h-2v7h6v-2h-4z"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Manrope-Regular",
                      "fontSize": 31,
                      "marginLeft": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                10
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#222149",
                  "borderRadius": 16,
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "maxHeight": 150,
                  "minWidth": 150,
                  "paddingBottom": 4,
                  "paddingLeft": 4,
                  "paddingRight": 4,
                  "paddingTop": 4,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Manrope-SemiBold",
                    "fontSize": 35,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Vento
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": "100%",
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={47}
                bbWidth={47}
                fill="#FFFFFF"
                focusable={false}
                height={47}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {},
                    ],
                    Object {
                      "flex": 0,
                      "height": 47,
                      "width": 47,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={47}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M0 0h24v24H0z"
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M10.5 17H4v-2h6.5a3.5 3.5 0 11-3.278 4.73l1.873-.703A1.5 1.5 0 1010.5 17zM5 11h13.5a3.5 3.5 0 11-3.278 4.73l1.873-.703A1.5 1.5 0 1018.5 13H5a3 3 0 010-6h8.5a1.5 1.5 0 10-1.405-2.027l-1.873-.702A3.501 3.501 0 0117 5.5 3.5 3.5 0 0113.5 9H5a1 1 0 100 2z"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Manrope-Regular",
                      "fontSize": 31,
                      "marginLeft": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                50
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#222149",
                  "borderRadius": 16,
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "maxHeight": 150,
                  "minWidth": 150,
                  "paddingBottom": 4,
                  "paddingLeft": 4,
                  "paddingRight": 4,
                  "paddingTop": 4,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Manrope-SemiBold",
                    "fontSize": 35,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Umidade
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": "100%",
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={47}
                bbWidth={47}
                fill="#FFFFFF"
                focusable={false}
                height={47}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {},
                    ],
                    Object {
                      "flex": 0,
                      "height": 47,
                      "width": 47,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={47}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M0 0h24v24H0z"
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M5.636 6.636L12 .272l6.364 6.364a9 9 0 11-12.728 0zM12 3.101L7.05 8.05A6.978 6.978 0 005 13h14a6.978 6.978 0 00-2.05-4.95L12 3.1z"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Manrope-Regular",
                      "fontSize": 31,
                      "marginLeft": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                50
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#222149",
                  "borderRadius": 16,
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "maxHeight": 150,
                  "minWidth": 150,
                  "paddingBottom": 4,
                  "paddingLeft": 4,
                  "paddingRight": 4,
                  "paddingTop": 4,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Manrope-SemiBold",
                    "fontSize": 35,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Visibilidade
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": "100%",
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={47}
                bbWidth={47}
                fill="#FFFFFF"
                focusable={false}
                height={47}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {},
                    ],
                    Object {
                      "flex": 0,
                      "height": 47,
                      "width": 47,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={47}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M0 0h24v24H0z"
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M1.181 12C2.121 6.88 6.608 3 12 3c5.392 0 9.878 3.88 10.819 9-.94 5.12-5.427 9-10.819 9-5.392 0-9.878-3.88-10.819-9zM12 17a5 5 0 100-10 5 5 0 000 10zm0-2a3 3 0 110-6 3 3 0 010 6z"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Manrope-Regular",
                      "fontSize": 31,
                      "marginLeft": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                visibility
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#222149",
                  "borderRadius": 16,
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "maxHeight": 150,
                  "minWidth": 150,
                  "paddingBottom": 4,
                  "paddingLeft": 4,
                  "paddingRight": 4,
                  "paddingTop": 4,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Manrope-SemiBold",
                    "fontSize": 35,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Nascer do sol
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": "100%",
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={47}
                bbWidth={47}
                fill="#FFFFFF"
                focusable={false}
                height={47}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {},
                    ],
                    Object {
                      "flex": 0,
                      "height": 47,
                      "width": 47,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={47}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M0 0h24v24H0z"
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M12 18a6 6 0 110-12 6 6 0 010 12zM11 1h2v3h-2V1zm0 19h2v3h-2v-3zM3.515 4.929l1.414-1.414L7.05 5.636 5.636 7.05 3.515 4.93zM16.95 18.364l1.414-1.414 2.121 2.121-1.414 1.414-2.121-2.121zm2.121-14.85l1.414 1.415-2.121 2.121-1.414-1.414 2.121-2.121zM5.636 16.95l1.414 1.414-2.121 2.121-1.414-1.414 2.121-2.121zM23 11v2h-3v-2h3zM4 11v2H1v-2h3z"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Manrope-Regular",
                      "fontSize": 31,
                      "marginLeft": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                sunrise
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#222149",
                  "borderRadius": 16,
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "maxHeight": 150,
                  "minWidth": 150,
                  "paddingBottom": 4,
                  "paddingLeft": 4,
                  "paddingRight": 4,
                  "paddingTop": 4,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Manrope-SemiBold",
                    "fontSize": 35,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Pôr do sol
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": "100%",
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={47}
                bbWidth={47}
                fill="#FFFFFF"
                focusable={false}
                height={47}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {},
                    ],
                    Object {
                      "flex": 0,
                      "height": 47,
                      "width": 47,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={47}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M0 0h24v24H0z"
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M9.984 5.06a6.5 6.5 0 1111.286 6.436A5.5 5.5 0 0117.5 21L9 20.999a8 8 0 11.984-15.94zm2.071.544a8.026 8.026 0 014.403 4.495 5.529 5.529 0 013.12.307 4.5 4.5 0 00-7.522-4.802z"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Manrope-Regular",
                      "fontSize": 31,
                      "marginLeft": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                sunset
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#222149",
                  "borderRadius": 16,
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "maxHeight": 150,
                  "minWidth": 150,
                  "paddingBottom": 4,
                  "paddingLeft": 4,
                  "paddingRight": 4,
                  "paddingTop": 4,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Manrope-SemiBold",
                    "fontSize": 35,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Sensação térmica
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": "100%",
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={47}
                bbWidth={47}
                fill="#FFFFFF"
                focusable={false}
                height={47}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {},
                    ],
                    Object {
                      "flex": 0,
                      "height": 47,
                      "width": 47,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={47}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M0 0h24v24H0z"
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M8 10.255V5a4 4 0 118 0v5.255a7 7 0 11-8 0zM8 16a4 4 0 108 0H8z"
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Manrope-Regular",
                      "fontSize": 31,
                      "marginLeft": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                16
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</BVLinearGradient>
`;
